### PR TITLE
cmd/evm: make t8ntool handle transaction decoding errors better

### PR DESCRIPTION
**Pull request by holiman**
This PR closes https://github.com/ethereum/go-ethereum/issues/27730 . By using an iterator instead of a slice of transactions, we can better handle the case when an individual transaction (within an otherwise well-formed RLP-list) cannot be decoded. 

For reviewers: 
- First commit just moves existing code into separate file (no change)
- Second commit just moves existing functionality from slice-form into iterator-form (no functional change)
- Third commit introduces rlp-iterator with individual handling of rlp-errors (functional change) + adds tests 

cc @winsvega

### CODE REVIEW

**Comment by winsvega** [comment]
```
Error: Client didn't reject transaction: (0xdb696fdd98aad26d3444d688df1ca50dd5947ebd11d5d8f3ef6c70fac0f8334f) 
0x03f901b101800285012a05f200833d090094095e7baea6a6c7c4c2dfeb977efac326af552d87830186a000f85bf85994095e7baea6a6c7c4c2dfeb977efac326af552d87f842a00000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000010af8e7a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f0001a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f0002a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f0003a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f0004a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f0005a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f0006a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065f000701a04c6a25969991d996e7f40964ba872e8dc6e8acdc422af055d9f404180dd2746ca013b66a111e61fb3ad575d456a1d976d221436f403f342ae8e78c19b177705607
Test Expected: TR_BLOBLIST_OVERSIZE
```

good, now I confirm t8n rejecting invalid blob transactions but this one still got through. 
maximum allowed blobs.

**Comment by holiman** [comment]
@winsvega if that is indeed a bug, then it deserves it's own issue in the tracker. But where is the max blobs per transaction defined? I don't see it in https://eips.ethereum.org/EIPS/eip-4844

**Comment by winsvega** [comment]
it is not defined expicitly but by calculating the blob price. max blobs defined per block.  so one tx can have 2, another 3. 
but the whole block no more than 6 because of header value calculations

**Comment by holiman** [comment]
Ok, pushed a fix, PTAL

**Comment by winsvega** [comment]
```
Error: Client reject transaction expected to be valid: (0x6f26856255f46b27b31d06e00750d8d75067fd8a28e15e8f5557a33fba288cb5) 
0x03f9010c01800285012a05f200833d090094095e7baea6a6c7c4c2dfeb977efac326af552d87830186a000f85bf85994095e7baea6a6c7c4c2dfeb977efac326af552d87f842a00000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000010af842a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8a001a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d880a0bf751ed5c37bd65d3ace5b73a1c62f7388b203a82ce366392e7b76fd2de12cb1a06f2b5344e5b997d35b3a0768006196a65c4eff8ed3acad5201a105c2b59b4e8c
Reason: blob gas (262144) would exceed maximum allowance 786432
```

false rejections allover the correct blob transactions. (correct by nimbus)

**Comment by winsvega** [comment]
now works!

File: cmd/evm/internal/t8ntool/tx_iterator.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 The go-ethereum Authors
```
Comment (MariusVanDerWijden) [inline cmd/evm/internal/t8ntool/tx_iterator.go:1]:
```suggestion
// Copyright 2023 The go-ethereum Authors
```

**Pull request merged**